crates/nargo_cli/src/cli/codegen_verifier_cmd.rs

```diff
@@ -1,23 +1,52 @@
-use super::fs::{create_named_dir, write_to_file};
+use super::fs::{
+    create_named_dir, keys::fetch_pk_and_vk, program::read_program_from_file, write_to_file,
+};
 use super::NargoConfig;
-use crate::{cli::compile_cmd::compile_circuit, constants::CONTRACT_DIR, errors::CliError};
-use acvm::SmartContract;
+use crate::{
+    cli::compile_cmd::compile_circuit, constants::CONTRACT_DIR, constants::TARGET_DIR,
+    errors::CliError,
+};
+use acvm::{ProofSystemCompiler, SmartContract};
 use clap::Args;
 use noirc_driver::CompileOptions;
 
 /// Generates a Solidity verifier smart contract for the program
 #[derive(Debug, Clone, Args)]
 pub(crate) struct CodegenVerifierCommand {
+    /// The name of the circuit build files (ACIR, proving and verification keys)
+    circuit_name: Option<String>,
```

Comment:
Is this changing the codegen verifier command? cc @TomAFrench

Reply:
In what way? It should be a non-breaking change as we''l fall back to the original behaviour if no circuit name is passed.